### PR TITLE
Add PuppetEnvironment keyword

### PR DESCRIPTION
**Pull request by avitova**

### CODE REVIEW

**Comment by lzap** [comment]
Snapshot test fails, to regenerate it, delete the file and run tests again:

    rm test/snapshots/foreman-web.json
    bundle exec rake test

It will generate the following change which you amend please:

```diff
diff --git a/test/snapshots/foreman-web.json b/test/snapshots/foreman-web.json
index 813685b..f43c275 100644
--- a/test/snapshots/foreman-web.json
+++ b/test/snapshots/foreman-web.json
@@ -795,6 +795,7 @@
         "Filebucket[puppet]"
       ],
       "keywords": [
+        "PuppetEnvironment:production",
         "PuppetHasCorrectiveChange",
         "PuppetHasSkips",
         "PuppetResourceFailed:File[mime_magic.conf]",
@@ -927,6 +928,7 @@
       }
     },
     "keywords": [
+      "PuppetEnvironment:production",
       "PuppetHasCorrectiveChange",
       "PuppetHasSkips",
       "PuppetResourceFailed:File[mime_magic.conf]",
```

TY!

**Comment by lzap** [comment]
We now have multiple tests, please rebase onto develop, delete all snapshots via `rm test/snapshots/*` and run tests once to regenerate them.

**Comment by ofedoren** [comment]
@avitova, I guess it needs a rebase.

**Comment by avitova** [comment]
> @avitova, I guess it need a rebase.

Looks better now, but I am not sure what was the intention in changing test/puppet_processor_test.rb. Should I leave it like this or delete it all?
Also in both my recent PRs there seems to be a problem with rufo:check.

File: test/puppet_processor_test.rb [hunk 1]
```diff
@@ -21,4 +21,110 @@ def test_keys_are_strings
       assert_snapshot(result, snapshot_filename)
     end
   end
+  
+  def test_deb
```
Comment (ofedoren) [inline test/puppet_processor_test.rb]: This was removed as part of https://github.com/theforeman/smart_proxy_host_reports/commit/ca6d93e2ac651af29ad38205d1eca6accd7a3313. I guess there is no need to return them.

I guess all the changes here should not be in this PR.

File: test/snapshots/foreman-web.json [hunk 2]
```diff
@@ -0,0 +1,942 @@
+{
```
Comment (ofedoren) [inline test/snapshots/foreman-web.json]: This file should not be present here...

**Review by ofedoren — COMMENTED** [review]
Thanks, @avitova, but there are few more issues, please see inline.

Regarding rufo: you can try to run `bundle exec rake rufo:run .` to automatically fix the issues.

**Comment by lzap** [comment]
Please rebase.

**Review by lzap — APPROVED** [review]

**Comment by lzap** [comment]
Thanks merged.

**Pull request merged**